login-utils/last.c

```diff
@@ -521,27 +547,39 @@ static int list(const struct last_control *ctl, struct utmpx *p, time_t logout_t
 	if (r < 0)
 		mem2strcpy(domain, p->ut_host, sizeof(p->ut_host), sizeof(domain));
 
+	/*
+	 * set last displayed character to an ellipsis '…'
+	 * when user/domain/ip fields are truncated in non-fullnames mode
+	 */
+	int fmt_dl = ctl->domain_len;
+	int fmt_nl = ctl->name_len;
+	if (!ctl->fullnames_mode) {
+		truncate_field(domain, sizeof(domain), LAST_DOMAIN_LEN, &fmt_dl);
+		truncate_field(p->ut_user, sizeof(p->ut_user), LAST_LOGIN_LEN, &fmt_nl);
+	}
+
 	if (ctl->showhost) {
 		if (!ctl->altlist) {
 			len = snprintf(final, sizeof(final),
 				"%-8.*s%c%-12.12s%c%-16.*s%c%-*.*s%c%-*.*s%c%s\n",
-				ctl->name_len, p->ut_user, ctl->separator, utline, ctl->separator,
```

Comment:
Something like this seems nicer:
```patch
--- a/login-utils/last.c
+++ b/login-utils/last.c
@@ -523,6 +523,9 @@ static int list(const struct last_control *ctl, struct utmpx *p, time_t logout_t
 
        if (ctl->showhost) {
                if (!ctl->altlist) {
+                       if (strlen(domain) > ctl->domain_len)
+                               domain[ctl->domain_len - 1] = '_';
+
                        len = snprintf(final, sizeof(final),
                                "%-8.*s%c%-12.12s%c%-16.*s%c%-*.*s%c%-*.*s%c%s\n",
                                ctl->name_len, p->ut_user, ctl->separator, utline, ctl->separator,
```

Reply:
If ellipsis were a single-byte char, that's what I would have done.

Reply:
done

Reply:
not a github person, so i don't know if i was supposed to resolve this conversation or not.

Reply:
If there is nothing left to discuss, it can be resolved.
Seems good in this case.